docs/components/emails.rst

```diff
@@ -1,13 +1,434 @@
+- Monitored Inbox Integration
+- Email transport/Email providers
+- Email stat helpers
+
+Email tokens and A/B testing
+----------------------------
+
+Email tokens are placeholders that you can insert into an Email.
+They get replaced by dynamic content once the Email gets sent or viewed in the browser.
+
+You can find examples of both Email token handling and A/B testing in the code example below.
+Both leverage the ``\Mautic\EmailBundle\EmailEvents::EMAIL_ON_BUILD`` event. Read more about listeners and subscribers (TODO add link).
+
+Email token capabilities consist of two parts: registering your custom tokens and rendering them.
+
+- ``$event->addToken($uniqueId, $htmlContent)`` allows you to show the email token in the email builder, so that users can easily add the token to their emails.
+- ``$event->getContent()`` and ``$event->setContent()`` are used for replacing the email token with actual dynamic content once the Email gets send or viewed in the browser.
+
+While Mautic supports :xref:`A/B testing` out of the box, you might have more complex needs to determine A/B test winner criteria.
+
+- ``$event->addAbTestWinnerCriteria()`` allows you to do exactly that. Using your custom logic, you can decide the winner of such criteria. An example is shown below.
+- ``$event->setAbTestResults()`` is where you set the actual A/B test results. More details are in the code example below.
+
+.. code-block:: PHP
+
+    <?php
+    // plugins/HelloWorldBundle/EventListener/EmailSubscriber.php
+
+    declare(strict_types=1);
+
+    namespace MauticPlugin\HelloWorldBundle\EventListener;
+
+    use Mautic\CoreBundle\Helper\TemplatingHelper;
+    use Mautic\EmailBundle\EmailEvents;
+    use Mautic\EmailBundle\Event\EmailBuilderEvent;
+    use Mautic\EmailBundle\Event\EmailSendEvent;
+    use MauticPlugin\HelloWorldBundle\HelloWorldEvents;
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+    class EmailSubscriber implements EventSubscriberInterface
+    {
+        private TemplatingHelper $templating;
+
+        public function __construct(TemplatingHelper $templating)
+        {
+            $this->templating = $templating;
+        }
+
+        public static function getSubscribedEvents()
+        {
+            return array(
+                EmailEvents::EMAIL_ON_BUILD   => array('onEmailBuild', 0),
+                EmailEvents::EMAIL_ON_SEND    => array('onEmailGenerate', 0),
+                EmailEvents::EMAIL_ON_DISPLAY => array('onEmailGenerate', 0)
+            );
+        }
+
+        /**
+        * Register the token and a custom A/B test winner
+        */
+        public function onEmailBuild(EmailBuilderEvent $event): void
+        {
+            // Displays the token in the email builder, so that users can easily find it and add it to their emails
+            $event->addToken('helloworld.token', 'Hello world token');
+
+            // Add AB Test Winner Criteria
+            $event->addAbTestWinnerCriteria(
```

Comment:
The old docs had a `callback` array key in `addAbTestWinnerCriteria`, but [it was removed in M3](https://github.com/mautic/mautic/pull/8145). Changed it to `event` and added some in-code documentation for it in https://github.com/mautic/mautic/pull/11047